otter/scheduler.py

```diff
@@ -19,81 +19,47 @@
 from otter.util.hashkey import generate_transaction_id
 
 
-class SchedulerService(TimerService):
+class SchedulerService(MultiService, object):
```

Comment:
`child.setServiceParent(parent)` calls `parent.addService(child)`. Circular references! Only MultiServices can be parents (or anything else implementing IServiceCollection).